src/vm/database/sqlite.rs

```diff
@@ -32,30 +32,92 @@ fn sqlite_has_entry(conn: &Connection, key: &str) -> bool {
     sqlite_get(conn, key).is_some()
 }
 
-impl KeyValueStorage for SqliteConnection {
-    fn put(&mut self, key: &str, value: &str) {
+impl SqliteConnection {
+    pub fn put(&mut self, key: &str, value: &str) {
         sqlite_put(&self.conn, key, value)
     }
 
-    fn get(&mut self, key: &str) -> Option<String> {
+    pub fn get(&mut self, key: &str) -> Option<String> {
         sqlite_get(&self.conn, key)
     }
 
-    fn has_entry(&mut self, key: &str) -> bool {
+    pub fn insert_metadata(&mut self, bhh: &BlockHeaderHash, contract_hash: &str, key: &str, value: &str) -> bool {
```

Comment:
What are the consequences and implications of this method returning `false`?

Reply:
Addressed in 505a1cd